### DISCUSSION TITLE
auto gen ui.ts

### DISCUSSION THREAD

**Discussion by wanghaisheng** [withastro/roadmap#1086] [Proposal]
### Body

```
export const languages = {
  en: 'English',
  fr: 'Français',
};

export const defaultLang = 'en';

export const ui = {
  en: {
    'nav.home': 'Home',
    'nav.about': 'About',
    'nav.twitter': 'Twitter',
  },
  fr: {
    'nav.home': 'Accueil',
    'nav.about': 'À propos',
  },
} as const;
```

currently  i dont find a way to auto gen this file ,need manually copy and paste,

**Comment by florian-lefebvre** [comment]
Why do you to auto-generate this file? From what data? In what context? Everything is possible but there's not enough info to help you currently

**Reply by wanghaisheng** [reply]
for any eixsting astro project ,we may have component pages with .astro, it cost time to one by one manually find the lable  use t(x.y.z) to replace them, never mind

**Reply by wanghaisheng** [reply]
by the way  there is bunch of work like  next+i18n+auto translation work over github . but you certain has plans

**Reply by florian-lefebvre** [reply]
Just to make sure I understand correctly, what you want is a way to extract text from .astro files into json files or equivalent to ease the i18n process. Is that right?

**Reply by wanghaisheng** [reply]
yes.I personally think it do has benefit for community. But  due to limit of knowledge I can not do such abstraction way

**Comment by RayyanNafees** [comment]
U should try [magicast](https://unjs.io/packages/magicast) to manually generate it using its ast

**Comment by florian-lefebvre** [comment]
I think I can safely say there are no plans to have this feature in core, but I'm sure that can be implemented as a community project (integration, cli etc...). Feel free top hop by the `#integrations` channel on our Discord server!
